### PR TITLE
feat: kodaprice

### LINKED ISSUE #7675: Failed to fetch price from Coingecko
image

### LINKED ISSUE #7992: Cache price from Coingecko in workers to mitigate cors
Seems the transfer doesn't work on canary, and it looks like on Coingecko error CORS, probably we should make some worker cache this price?

image

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Hit to kodaprice first, if error fallback to coingecko
- [x] Fix #7992
- [x] Fix #7675

#### Had issue bounty label?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

### <samp>🤖 Generated by Copilot at 70953d4</samp>

Refactored the code for fetching and displaying token prices in the gallery. Moved the price-related functions from `utils/coingecko.ts` and `utils/support.ts` to a new file `utils/coinprice.ts` that supports different tokens and APIs. Updated the components and store that use these functions to import them from the new file. Added the `kodaprice` API as a new source of token prices.

### <samp>🤖 Generated by Copilot at 70953d4</samp>

> _To fetch and format token prices_
> _We moved some functions and renamed devices_
> _We used `ofetch` and `kodaprice`_
> _And made the code more concise_
> _Now we can support more tokens and currencies_

### CODE REVIEW

File: utils/coingecko.ts
Comment (preschian) [inline utils/coingecko.ts:1]: Actually, I renamed `coingecko.ts` to `coinprice.ts` 🙏

**Comment by yangwao** [comment]
Missing those alt sources and fallback?
- https://github.com/kodadot/nft-gallery/issues/7675#issuecomment-1764345212

**Review by daiagi — APPROVED** [review]
Code lgtm

**Review by Jarsen136 — APPROVED** [review]

**Comment by prury** [comment]
was able to reproduce CORS on canary by refreshing transfer page many times
same did not occur when testing this deployed version, nice! 

image

**Comment by yangwao** [comment]
Thanks!
Let's resolve conflicts and merge it @preschian 
pay 100 USD

---
including
- https://github.com/kodadot/workers/pull/195

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 5.27 USD/DOT ~ 18.975 $DOT
🧗 [1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p](https://polkadot.subscan.io/account/1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)
🔗 [0x7e6e1e417b224bf9787cf19287615d31f5854cf09053f28caa7bb30133f34064](https://polkadot.subscan.io/extrinsic/0x7e6e1e417b224bf9787cf19287615d31f5854cf09053f28caa7bb30133f34064)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**